ERCS/erc-7795.md

```diff
@@ -0,0 +1,589 @@
+        "version": "1.0",
+        "chainId": "0x01",
+        "owner": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "token": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+        "tokenId": "0x10",
+        "minAmount": "0x0f4240"
+      }
+    }
+  }
+]
+```
+
+### ERC-1155 Minimum Allowance Capability
+
+A dApp can use the `erc1155MinAllowance` capability in a `wallet_sendCalls` request to request that a wallet ensure the owner has a minimum allowance of a specified [ERC-1155](./eip-1155.md) token.
+Note this capability does not imply that the owner has a balance of the token, only that the allowance is equal to or greater than the specified amount.
+
+#### `wallet_getCapabilities` Response
+
+Schema:
+
+```typescript
+type Erc1155MinAllowanceCapability = {
+  supported: boolean;
+  versions: string[];
+}
+```
+
+Example:
+
+```json
+{
+  "0x1": {
+    "erc1155MinAllowance": {
+      "supported": true,
+      "versions": ["1.0"]
+    }
+  }
+}
+```
+
+#### `wallet_sendCalls` Request
+
+Version 1.0 Schema:
+
+```typescript
+type Erc1155MinAllowanceParams = {
+  version: string;
+  chainId: `0x${string}`; // Hex chain id
+  owner: `0x${string}`; // Address
+  operator: `0x${string}`; // Address
+  token: `0x${string}`; // Address
+  tokenId: `0x${string}`; // Hex value
+  minAmount: `0x${string}`; // Hex value
+};
+```
+
+This capability requests that the `owner` address **MUST** have an allowance of at least `minAmount` of `tokenId` of `token` for `operator` on `chainId` before the transaction is executed.
+
+Example:
+
+```json
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "calls": [
+      {
+        "to": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+        "value": "0x00",
+        "data": "0x...",
+        "chainId": "0x01",
+      }
+    ],
+    "capabilities": {
+      "erc1155MinAllowance": {
+        "version": "1.0",
+        "chainId": "0x01",
+        "owner": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "operator": "0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d",
+        "token": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+        "tokenId": "0x10",
+        "minAmount": "0x0f4240"
+      }
+    }
+  }
+]
+```
+
+## Rationale
+
+This ERC extends [EIP-5792](./eip-5792.md) rather than defining new RPC methods because:
+
+1. **Consistency**: Leverages existing capability discovery mechanism
+2. **Composability**: Requirements can be combined with other [EIP-5792](./eip-5792.md) capabilities
+3. **Flexibility**: Wallets can implement only the requirements they support
+4. **Extensibility**: New requirement types can be added as additional capabilities
+
+The decision to split requirements into individual capabilities rather than a single capability type allows:
+
+1. Granular support by wallets
+2. Clear capability discovery
+3. Independent versioning of requirement types
+4. Simpler implementation for basic wallets
+
+## Reference Implementation
```

Comment:
The reference implementation section should contain a minimal example implementation of the standard, not usage examples. This content belongs in the specification section, especially considering it's introducing new requirements (with MUST/MAY/etc.)